### ISSUE TITLE
Plugin: Datalist

### ISSUE THREAD

**Issue by jvanulde** [wet-boew/wet-boew#6486]
I am loading data dynamically into the datalist plugin and it works fine in all browsers except Safari. It appears that the list is not destroyed between updates. See screenshot below. I am using a slightly modified version of the code sample provided:

		var mapDiv = $("#geomap-map-" + geomap.mapid),
			selector = "#wb-geomap-geocode-search-" + geomap.mapid,
			xhr,
			timer,
			val,
			keys, keycode;

		mapDiv.append(
			"<div class='geomap-geoloc'>" +
				"<label for='wb-geomap-geocode-search-" + geomap.mapid + "' class='wb-inv'>" + i18nText.geoCoderLabel + "</label>" +
				"<input type='text' class='form-control input-sm opct-90' name='wb-geomap-geocode-search-" + geomap.mapid + "' id='wb-geomap-geocode-search-" + geomap.mapid + "' list='wb-geomap-geocode-results-" + geomap.mapid + "' autocomplete='off' placeholder='" + i18nText.geoCoderPlaceholder + "' />" +
				"<datalist id='wb-geomap-geocode-results-" + geomap.mapid + "'></datalist>" +
			"</div>"
		);

		$document.on( "keyup", selector, function( evt ) {

			var $elm = $( this );

			keys = [ 9, 13, 16, 17, 18, 19, 20, 33, 34, 35, 36, 37, 38, 39, 40, 45, 224 ];

			val = $elm.val();

			keycode = evt.which;

			if ( val === "" || val.length <= 2  || $.inArray( parseInt( keycode ), keys ) > -1 ) {
				return;
			}

			$( "#wb-geomap-geocode-results-" + geomap.mapid ).empty();

			clearTimeout( timer );

			timer = setTimeout(	function() {
				$elm.trigger({
					type: "ajax-fetch.wb",
					fetch: {
						url: encodeURI( i18nText.geoLocationURL + "?q=" + val + "*" ),
						dataType: wb.ielt10 ? "jsonp" : "json",
						jsonp: wb.ielt10 ? "callback" : null
					}
				});
			}, 500 );

		});

		$document.on( "ajax-fetched.wb", selector, function( event ) {

			var items = wb.ielt10 ? event.fetch.response.data : event.fetch.response,
				$dataList = $( "#wb-geomap-geocode-results-" + geomap.mapid ),
				lenItems = items.length,
				options = "",
				indItem, item, bnd, ll, title;

			$dataList.empty();

			for ( indItem = 0; indItem !== lenItems; indItem += 1 ) {

				item = items[ indItem ];

				title = item.title
					.replace(/&/g, "&amp;")
					.replace(/"/g, "&quot;")
					.replace(/'/g, "&#39;")
					.replace(/</g, "&lt;")
					.replace(/>/g, "&gt;");

				bnd = item.bbox ? item.bbox[ 0 ] + ", " + item.bbox[ 1 ] + ", " + item.bbox[ 2 ] + ", " + item.bbox[ 3 ] : null;
				ll = item.geometry && item.geometry.type === "Point" ? item.geometry.coordinates[ 0 ] + ", " + item.geometry.coordinates[ 1 ] : null;

				options += "<option label=\"" + title + "\" value=\"" + title + "\" + data-lat-lon=\"" + ll + "\" data-bbox=\"" + bnd  + "\" data-type=\"" + item.type + "\"></option>";
			}

			if ( wb.ielt10 ) {
				options = "<select>" + options + "</select>";
			}

			$dataList.append( options );

			$( this ).trigger( "wb-update.wb-datalist" );
		});

![screen shot 2015-01-29 at 6 07 18 pm](https://cloud.githubusercontent.com/assets/2255248/5970082/1225f288-a7e2-11e4-90cc-1730252e4795.png)

**Comment by biljanap** [comment]
I believe <code>datalist</code> element is not supported in some browsers. Unfortunately Safary is one of them.
<a href="http://www.w3schools.com/tags/tag_datalist.asp">W3schools: HTML datalist element</a>
<a href="http://caniuse.com/#search=datalist">Can I Use: datalist</a>

**Comment by pjackson28** [comment]
@biljanap We provide a polyfill to add support for browsers that lack native datalist support.

**Comment by biljanap** [comment]
@pjackson28 Is this being used for Search Canada.ca?

**Comment by pjackson28** [comment]
@biljanap Yes it is. Here is a video of it working on Safari 7: https://saucelabs.com/tests/c1fea71c96984349bbeb93d2544b2dc9

Note that the incredible lagginess is from the really slow cloud-based testing. It eventually works, but the lag made it much more difficult to select an item in the datalist (wouldn't be an issue when dealing directly with Safari).

**Comment by biljanap** [comment]
@pjackson28 I was checking search functionality on Canada site using Safary on my iPhone. It's not working. Is it supposed to work for mobiles too?

**Comment by pjackson28** [comment]
@biljanap It looks like it is an issue with loading initially in mobile view rather than using Safari. It looks like the dynamic datalist support used by Canada.ca is not loading in that case. Oddly enough it loads fine if you start with desktop view and then switch to mobile view by shrinking the browser viewport.

@LaurentGoderre Please look into this. The menu plugin takes care of initializing the overlay and the details/summary elements here: https://github.com/wet-boew/wet-boew/blob/master/src/plugins/menu/menu.js#L315-L319. How should the dynamic datalist implementation from Canada.ca be handled here?

**Comment by jvanulde** [comment]
I found a workaround by removing the wb-al-cnt on each load/refresh:

    // BUG: remove the content div for Safari
    // since the datalist plugin doesn't do this
    dataList.parent().find( ".wb-al-cnt").remove();

**Comment by LaurentGoderre** [comment]
Interesting maybe we can look at adding this to the polyfill

**Comment by hsrudnicki** [comment]
Can this issue be closed or otherwise?

**Comment by hsrudnicki** [comment]
@LaurentGoderre @jvanulde and others: Have been the code update added to the polyfill, if yes then this issue can be closed or otherwise? Either way, please update this issue's status, thank you

**Comment by hsrudnicki** [comment]
Is this still an issue? Either way, please update again this issue's status, whether this issue can be closed as resolved/fixed or otherwise, thank you

**Comment by hsrudnicki** [comment]
Does this remain still an issue? Please update the status of this, whether this can be closed or otherwise, thank you
